dev/static_magic.h

```diff
@@ -9,22 +9,22 @@ namespace sqlite_orm {
     namespace internal {
 
         template<typename T, typename F>
-        auto static_if(std::true_type, T t, F) {
-            return std::move(t);
+        decltype(auto) static_if(std::true_type, const T &t, const F &) {
```

Comment:
callable object must not be copied or moved within `static_if` cause it affects its state